src/main/java/co/rsk/federate/signing/hsm/message/AdvanceBlockchainMessage.java

```diff
@@ -1,19 +1,23 @@
 package co.rsk.federate.signing.hsm.message;
 
 import java.util.ArrayList;
+import java.util.Collections;
 import java.util.List;
+
 import org.ethereum.core.BlockHeader;
 import org.spongycastle.util.encoders.Hex;
 
 public class AdvanceBlockchainMessage {
     private final List<String> blockHeaders;
+    private final List<String[]> brothersList;
 
-    public AdvanceBlockchainMessage(List<BlockHeader> blockHeaders) {
+    public AdvanceBlockchainMessage(List<BlockHeader> blockHeaders, List<List<BlockHeader>> brothersList) {
         this.blockHeaders = new ArrayList<>();
         // Invert order
         for (int index = blockHeaders.size() - 1; index >= 0; index--) {
             this.blockHeaders.add(parseBlockHeader(blockHeaders.get(index)));
         }
+        this.brothersList = Collections.unmodifiableList(parseBrothersBlockHeaders(brothersList));
```

Comment:
The block headers list is still being exposed. Should we also protect it?

Reply:
Resolved